### ISSUE TITLE
statusline: enable %@ (clickable regions, like 'tabline')

### ISSUE THREAD

**Issue by wbthomason** [neovim/neovim#11924]
- `nvim --version`:
```
NVIM v0.5.0-389-g9897ad360
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/cc -march=native -O2 -pipe -fno-plt -O2 -DNDEBUG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/tmp/yaytmp-1000/neovim-git/src/neovim-git/build/config -I/tmp/yaytmp-1000/neovim-git/src/neovim-git/src -I/usr/include -I/tmp/yaytmp-1000/neovim-git/src/neovim-git/build/src/nvim/auto -I/tmp/yaytmp-1000/neovim-git/src/neovim-git/build/include
Compiled by wil@loki

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

### Actual behaviour
Per `:h statusline`, `%@` enables clickable tabline regions that run arbitrary functions.

### Expected behaviour
It would be useful for `%@` to work for both the tabline *and* the statusline.

**Comment by andymass** [comment]
What is the example use case?

**Comment by wbthomason** [comment]
Here are a few examples:
- Suppose I have a statusline component showing, e.g., LSP client information. Some client parameters (e.g. choice of interpreter for Python, etc.) may be configurable, and it would be nice if I could click on the segment to bring up a menu where I can change their values.
- More simply, imagine I have some component showing the value of some setting, or some process (e.g. a linter) that I may want to toggle between values/toggle on and off. It would be convenient to be able to toggle this with a click.
- I may want to be able to click my linter status component to toggle the quickfix list, or click git status information to open a branch or commit selector, etc.
- I could imagine this enabling a "notifications center"-type statusline component that, when clicked, shows a floating window with the contents of `:messages` since the last time it was clicked (or some other set of notifications/messages).

These are all things I can do pretty conveniently already with a command or keybinding. I wouldn't argue there's anything _uniquely_ feasible by clicking on the statusline. It would, however, still be nice to have in use cases like the above.

**Comment by iasj** [comment]
This would be perfect for my [plugin](https://gitlab.com/nvpm/nvpm2).

As you can see, I use the tabline as well as the statusline to show the `workspace`, the `tabs` and the `buffers`, considering the plugin's terminology. 

I've been thinking on implementing mouse events for its version 3, but now I think I might wait for this feature. 

Any idea when it will be ready, if ever?

**Comment by zeertzjq** [comment]
Implemented in #18650
